### PR TITLE
feat: add Chapter 7

### PR DESCRIPTION
**Pull request by skylee03**

### CODE REVIEW

File: docs/07-imperative-programming.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+the `alter` function takes two mutable references to `Ref` structs, and then changes the value of the `val` field of both of them. This means that the `x` variable will also be changed twice, because it's just an alias for the `ref` variable.
+
+## Debugger
+
+Debugging is like being a detective. You're trying to figure out why your program isn't working as expected. MoonBit's debugger is a tool that helps you do this by showing you what's happening inside your program as it runs. You can pause the program at any point, look at the values of variables, and step through the code one line at a time. This is incredibly useful for understanding complex behavior and fixing bugs.
+
+![debugger](/pics/debugger.webp)
+
+## Loops
+
```
Comment (peter-jerry-ye) [inline docs/07-imperative-programming.md:121]: Have we introduced for-else loop and functional loop yet?

**Comment by peter-jerry-ye** [comment]
Also check with the original script to see if anything is missing

File: docs/07-imperative-programming.md (hunk 1 above)
Comment (skylee03) [reply docs/07-imperative-programming.md:121]: We used functional loop in [Chapter 4](https://moonbitlang.github.io/moonbit-textbook/tuples-structs-enums), but we haven't introduced them yet.

Comment (skylee03) [reply docs/07-imperative-programming.md:121]: Maybe I need to update the example to avoid introducing the loops too early.

Comment (peter-jerry-ye) [reply docs/07-imperative-programming.md:121]: Yeah, I think we should remove the usage of functional loop there. I also notice that we need to introduce function matrix in Chapter 5 just after pattern matching.

**Comment by skylee03** [comment]
I have added some content that was in the Chinese lecture scripts but is missing in the current English version, but have not introduced any new content yet.

**Pull request merged**